sdk/include/opentelemetry/sdk/metrics/instruments.h

```diff
@@ -19,6 +19,9 @@ enum class InstrumentType
   kCounter,
   kHistogram,
   kUpDownCounter,
+#if OPENTELEMETRY_ABI_VERSION_NO >= 2
+  kGauge,
+#endif
```

Comment:
It will be simpler to just declare this enum value all the time, and not conditionally.

The sdk headers can change, and having kGauge declared even in ABI v1 is not disruptive.

It will just be dead code in ABI v1.

Please adjust the switch/cases accordingly, to avoid the ifdef